app/webpacker/controllers/dirty_form_controller.js

```diff
@@ -0,0 +1,54 @@
+import ApplicationController from "./application_controller";
+const LEAVING_PAGE_MESSAGE = "Are you sure you want to leave?";
+
+export default class extends ApplicationController {
+  static values = { dirty: Boolean };
+  static targets = ["submitBtn"];
+
+  connect() {
+    // find all inputs within the form
+    // and attach eventListeners
+    this.inputs = this.element.querySelectorAll("input");
```

Comment:
I think I'd use `targets`

Reply:
Good catch on the `textarea` and `select`! This is why I wanted your input on this.

I think targets are a bit too cumbersome,( we have a large form in this case and might have other large forms ), but I can definitely see that we might require to `target`  some obscure custom element or something in the future, without bloating our list of elements that need eventListeners.

Let me experiment, and try to correct the errors that are present right now, see what I'll find, and make a decision on the `targets` then.